lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp

```diff
@@ -701,6 +702,139 @@ void TypeLoweringVisitor::visitExpr(SubindexOp op) {
+  // This records a map of bundle subfield name and the corresponding subfield
+  // operation.
+  DenseMap<StringRef, SubfieldOp> subfieldUsers;
+  BundleType vectorOfBundleType = vectorElementType.dyn_cast<BundleType>();
+
+  if (vectorOfBundleType) {
+    // If the vector element is not a bundle, then only one mux tree generated,
+    // else one mux tree for each bundle element. This loop initializes the mux
+    // tree with the first element of the vector (for each element of the
+    // bundle).
+    for (size_t elem = 1; elem < vectorOfBundleType.getNumElements(); elem++)
+      muxTreeVec.push_back(loweredVector[elem]);
+    // User of a vector of bundles is a SubfieldOp.
+    for (auto u : op->getUsers())
+      if (auto subField = dyn_cast<SubfieldOp>(u))
+        subfieldUsers[subField.fieldname()] = subField;
+  }
+
+  // Create a true constant.
+  Value constTrue = builder->create<ConstantOp>(
+      op.getLoc(), UIntType::get(op.getContext(), 1),
+      getIntAttr(APInt(1, 1), op.getContext()));
+
+  size_t dimRange = 0;
+  for (size_t elem = muxTreeVec.size(); elem < loweredVector.size();
+       elem += muxTreeVec.size()) {
+    Value isIndexEq = constTrue;
+    for (size_t ind = 0; ind < allDims.size(); ind++) {
+      // Get the index variable.
+      auto index = allDims[ind].index();
+      // Get the constnat int, with which to compare the index.
+      auto indexInt = indicesRange[ind];
+      auto selectWidth =
+          index.getType().cast<FIRRTLType>().getBitWidthOrSentinel();
+      // Create " && index == indexInt?"
+      isIndexEq = builder->create<AndPrimOp>(
+          UIntType::get(op.getContext(), 1), isIndexEq,
+          builder->create<EQPrimOp>(
+              op.getLoc(), UIntType::get(op.getContext(), 1), index,
+              builder->create<ConstantOp>(
+                  op.getLoc(), UIntType::get(op.getContext(), selectWidth),
+                  getIntAttr(APInt(selectWidth, indexInt), op.getContext()))));
+      // This is the logic to generate the range of constant integers, with
+      // which to compare the index.
+      if (ind == dimRange) {
+        size_t maxElems = 1 << selectWidth;
+        // Make sure, we donot perform out of bounds access here ?
+        indicesRange[ind] = (indicesRange[ind] + 1) % maxElems;
+        if (!indicesRange[ind])
+          dimRange++;
+        else
+          dimRange = 0;
+      }
+    }
+    // Generate the mux tree for each element of the bundle.
+    for (size_t m = 0; m < muxTreeVec.size(); m++)
+      muxTreeVec[m] = builder->create<MuxPrimOp>(
+          op.getLoc(), muxTreeVec[m].getType().cast<FIRRTLType>(), isIndexEq,
+          loweredVector[elem + m], muxTreeVec[m]);
+  }
+  if (vectorOfBundleType) {
+    size_t elemNum = 0;
+    for (auto bundleElem : vectorOfBundleType.getElements()) {
+      auto fieldName = bundleElem.name.getValue();
+      auto iter = subfieldUsers.find(fieldName);
+      if (iter != subfieldUsers.end()) {
```

Comment:
can the .find call here fail (return end)?  If so what does that mean?  Should this be an assertion?

Reply:
This can happen when one of the subfields of a bundle has no use. So it shouldn't be an error, and I think can be ignored.